src/telliot_feeds/sources/price/spot/curvefi.py

```diff
@@ -0,0 +1,106 @@
+from dataclasses import dataclass
+from dataclasses import field
+from typing import Any
+
+from telliot_feeds.dtypes.datapoint import datetime_now_utc
+from telliot_feeds.dtypes.datapoint import OptionalDataPoint
+from telliot_feeds.pricing.price_service import WebPriceService
+from telliot_feeds.pricing.price_source import PriceSource
+from telliot_feeds.utils.log import get_logger
+
+
+logger = get_logger(__name__)
+
+
+contract_map = {
+    "eth": "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE",
+    "steth": "0xae7ab96520DE3A18E5e111B5EaAb095312D7fE84",
+    "btc": "0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599",
+}
+
+
+class CurveFinanceSpotPriceService(WebPriceService):
+    """CurveFinance Price Service"""
+
+    def __init__(self, **kwargs: Any) -> None:
+        kwargs["name"] = "CurveFinance Price Service"
+        kwargs["url"] = "https://api.curve.fi"
+        super().__init__(**kwargs)
+
+    async def get_price(self, asset: str, currency: str) -> OptionalDataPoint[float]:
+        """This implementation gets the price from the Curve finance API."""
+        asset = asset.lower()
+        currency = currency.lower()
+        if asset not in contract_map:
+            logger.error(f"Asset not supported: {asset}")
+            return None, None
+        asset_address = contract_map[asset]
+
+        request_url = "/api/getPools/ethereum/main"
+
+        d = self.get_url(request_url)
+
+        if "error" in d:
+            logger.error(d)
+            return None, None
+        elif "response" in d:
+            response = d["response"]
+            data = response.get("data")
+
+            if data is None:
+                logger.error("No data in returned response")
+                return None, None
+            pool_data = data.get("poolData")
+            if pool_data is None:
+                logger.error("Failed to parse response data from Curve Finance API")
+                return None, None
+            asset_price = None
+            for pool in pool_data:
+                for coin in pool["coins"]:
+                    if coin["address"] == asset_address:
+                        asset_price = coin.get("usdPrice")
+                        break
+                else:
```

Comment:
I think this else statement is not needed

Reply:
I think the else statement makes the code more efficient because without it would keep iterating through the outer loop even after the price is found.  but ill change the code, I can see how this looks confusing